meteor/client/ui/SegmentTimeline/SegmentTimelinePart.tsx

```diff
@@ -428,533 +454,587 @@ const CARRIAGE_RETURN_ICON = (
+					),
+					SegmentTimelinePartClass.getLiveLineTimePadding(this.props.timeScale)
+				) *
+					this.props.timeScale +
+				'px',
+		}
+	}
+
+	private renderEndOfSegment = (t: TFunction, innerPart: Part, isEndOfShow: boolean, isEndOfLoopingShow?: boolean) => {
+		return (
+			<>
+				{this.props.isLastInSegment && (
+					<div
+						className={ClassNames('segment-timeline__part__nextline', 'segment-timeline__part__nextline--endline', {
+							'auto-next': innerPart.autoNext,
+							'is-next':
+								this.state.isLive &&
+								((!this.props.isLastSegment && !this.props.isLastInSegment) ||
+									!!this.props.playlist.nextPartInstanceId),
+							'show-end': isEndOfShow,
+						})}
+					>
 						<div
-							className={ClassNames('segment-timeline__part', {
-								live: this.state.isLive,
-								next: this.state.isNext || this.props.isAfterLastValidInSegmentAndItsLive,
-								invalid: innerPart.invalid && !innerPart.gap,
-								floated: innerPart.floated,
-								gap: innerPart.gap,
-								'invert-flash': this.state.highlight,
-
-								'duration-settling': this.state.isDurationSettling,
+							className={ClassNames('segment-timeline__part__nextline__label', {
+								'segment-timeline__part__nextline__label--thin': innerPart.autoNext && !this.state.isLive,
 							})}
-							data-obj-id={this.props.part.instance._id}
-							id={SegmentTimelinePartElementId + this.props.part.instance._id}
-							style={{ ...this.getLayerStyle(), ...invalidReasonColorVars }}
 						>
-							{innerPart.invalid ? <div className="segment-timeline__part__invalid-cover"></div> : null}
-							{innerPart.floated ? <div className="segment-timeline__part__floated-cover"></div> : null}
+							{innerPart.autoNext && t('Auto') + ' '}
+							{this.state.isLive && t('Next')}
+							{isEndOfLoopingShow && <LoopingIcon />}
+						</div>
+					</div>
+				)}
+				{!isEndOfShow && this.props.isLastInSegment && (
+					<div
+						className={ClassNames('segment-timeline__part__segment-end', {
+							'is-next':
+								this.state.isLive &&
+								((!this.props.isLastSegment && !this.props.isLastInSegment) ||
+									!!this.props.playlist.nextPartInstanceId),
+						})}
+					>
+						<div className="segment-timeline__part__segment-end__label">
+							<SegmentEnd />
+						</div>
+					</div>
+				)}
+				{isEndOfShow && !this.props.playlist.loop && (
+					<div className="segment-timeline__part__show-end">
+						<div className="segment-timeline__part__show-end__label">{t('Show End')}</div>
+					</div>
+				)}
+				{isEndOfShow && this.props.playlist.loop && (
+					<div className="segment-timeline__part__show-end loop">
+						<div className="segment-timeline__part__show-end__label">{t('Loops to top')}</div>
+					</div>
+				)}
+			</>
+		)
+	}
+
+	static convertHexToRgba(hexColor: string): { red: number; green: number; blue: number } | undefined {
```

Comment:
This should be named `convertHexToRgb`, since there is no alpha value returned :)